test/IntersectionUtils.test.js

```diff
@@ -157,21 +158,149 @@ describe( 'Triangle Intersections', () => {
 
-		expect( t1.intersectsTriangle( t2 ) ).toBe( true );
+		expect( t1.intersectsTriangle( t2 ) ).toBe( false );
+
+	} );
+
+} );
+
+describe( 'Triangle Intersection line', () => {
+
+	const t1 = new SeparatingAxisTriangle();
+	const t2 = new Triangle();
+	const target = new Line3();
+	const expected = new Line3();
+
+	const expectVerticesToBeClose = ( a, b ) => {
+
+		expect( a.x ).toBeCloseTo( b.x );
+		expect( a.y ).toBeCloseTo( b.y );
+		expect( a.z ).toBeCloseTo( b.z );
+
+	};
+
+	const expectLinesToBeClose = ( a, b ) => {
+
+		try {
+
+			expectVerticesToBeClose( a.start, b.start );
+			expectVerticesToBeClose( a.end, b.end );
+
+		} catch {
+
+			expectVerticesToBeClose( a.end, b.start );
+			expectVerticesToBeClose( a.start, b.end );
+
+		}
+
+	};
+
+	it( 'should intersect on point', () => {
+
+		t1.a.set( 0, 0, 0 );
+		t1.b.set( 0, 0, 2 );
+		t1.c.set( 2, 0, 0 );
+		t1.needsUpdate = true;
+
+		t2.a.set( 1, - 1, 0 );
+		t2.b.set( 1, 1, 0 );
+		t2.c.set( 1, 0, - 1 );
+
+		expected.start.set( 1, 0, 0 );
+		expected.end.set( 1, 0, 0 );
+
+		expect( t1.intersectsTriangle( t2, target ) ).toBe( true );
+		expectLinesToBeClose( target, expected );
 
 	} );
 
+	it( 'should intersect', () => {
+
+		t1.a.set( 0, 0, 0 );
+		t1.b.set( 0, 0, 5 );
+		t1.c.set( 5, 0, 0 );
+		t1.needsUpdate = true;
+
+		t2.a.set( 1, - 1, 1 );
+		t2.b.set( 1, - 1, - 1 );
+		t2.c.set( 1, 1, 1 );
+
+		expected.start.set( 1, 0, 0 );
+		expected.end.set( 1, 0, 1 );
+
+		expect( t1.intersectsTriangle( t2, target ) ).toBe( true );
+		expectLinesToBeClose( target, expected );
+
+	} );
+
+	it( 'should intersect on common side', () => {
+
+		t1.a.set( 0, 0, 0 );
+		t1.b.set( 3, 0, 0 );
+		t1.c.set( 0, 1, 2 );
+		t1.needsUpdate = true;
+
+		t2.a.set( 1, 0, 0 );
+		t2.b.set( 2, 0, 0 );
+		t2.c.set( 0, 1, - 2 );
+
+		expected.start.set( 1, 0, 0 );
+		expected.end.set( 2, 0, 0 );
+
+		expect( t1.intersectsTriangle( t2, target ) ).toBe( true );
+		expectLinesToBeClose( target, expected );
+
+	} );
+
+	// coplanar cases
+	it( 'should be coplanar and line is zero', () => {
+
+		t1.a.set( 0, 0, 0 );
+		t1.b.set( 3, 0, 0 );
+		t1.c.set( 0, 0, 2 );
+		t1.needsUpdate = true;
+
+		t2.a.set( 1, 0, 0 );
+		t2.b.set( 2, 0, 0 );
+		t2.c.set( 0, 0, - 2 );
+
+		expected.start.set( 0, 0, 0 );
+		expected.end.set( 0, 0, 0 );
+
+		expect( t1.intersectsTriangle( t2, target ) ).toBe( true );
+		expectLinesToBeClose( target, expected );
+
+	} );
+
+	it( 'triangles almost coplanar should intersect on point', () => {
+
+		t1.a.set( 0.0720, 0.2096, 0.3220 );
+		t1.b.set( 0.0751, 0.2148, 0.3234 );
+		t1.c.set( 0.0693, 0.2129, 0.3209 );
+		t1.needsUpdate = true;
+
+		t2.a.set( 0.0677, 0.2170, 0.3196 );
+		t2.b.set( 0.0607, 0.2135, 0.3165 );
+		t2.c.set( 0.0693, 0.2129, 0.3209 );
+
+		expected.start.set( 0.0693, 0.2129, 0.3209 );
+		expected.end.set( 0.0693, 0.2129, 0.3209 );
+
+		expect( t1.intersectsTriangle( t2, target ) ).toBe( true );
+		expectLinesToBeClose( target, expected );
+
+	} );
```

Comment:
@minitoine Given the nature of the floating point error and how small the triangles / intersection surface is here this intersection test now returns false and it no longer returns a nonsensical line result. If that sounds good to you I'll modify the test to reflect the behavior of the funciton.

Reply:
I guess this is _ok-ish_. You goal here is not to have a super-high precision triangles intersection algorithm.
Maybe could we scale up the triangles based on, i don't know, the area value under a threshold? 
In order to make the code more robust.
Anyway, it is better to return false than a wrong result!